### ISSUE TITLE
Support for anonymous types

### ISSUE THREAD

**Issue by AKTheKnight** [yanpaulo/RestHttpClient#4]
https://www.reddit.com/r/dotnet/comments/b2nufu/restful_api_access_made_simple_with/eiv5er9

Based on the reddit comment linked below, it would be good to add support for anonymous types such as below:

```csharp
var todo = new { Title = "", UserId = 0 };
var todoVal = await client.RestGetAnonymousAsync("todos/1", todo);
```

Not sure if you would like this to be done as:
- An overload like this:
```csharp
RestHttpClient#RestGetAsync<T>(string url, T type)
```

- Or as a new method:
```csharp
RestHttpClient#RestGetAnonymousAsync<T>(string url, T type)
```
Either way it'll mean adding a new method to the IDeserializer, not sure what your thoughts are on that?
```csharp
T DeserializeAnonymous<T>(string content, T type);
```

**Comment by yanpaulo** [comment]
Well timing is perfect because I'm implementing #5 right now.

It would be really nice to support anonymous types, but don't you think the interface would become a little more difficult to implement?

**Comment by AKTheKnight** [comment]
I think by using `T Deserialize<T>(string content, T type);` as the interface method, people can implement is as below if they wanted (This passes all the tests, I just keep it as a separate method so we take advantage of any possible library benefits to anonymous deseralization):
```csharp
        public T Deserialize<T>(string content, T type)
        {
            return Deserialize<T>(content);
        }
```

**Comment by yanpaulo** [comment]
Good news! There is no need to change the interface at all to implement this, as of c7916538cf687f25f89db6bf1242e6923283c9fe 
@AKTheKnight  I see that you've made PR #6  and most of these changes won't be necessary, and so I will take only the tests ok?

**Comment by AKTheKnight** [comment]
Yes of course, feel free to take whatever :)

**Comment by yanpaulo** [comment]
@AKTheKnight merged your PR with my own code and it's like this: ab94fd2da1f139dda0f116f3f2144140c5a94ed6
But if I merge it, you won't appear as a contributor. If you would like so, please fix PR #6  and I will merge it instead :)
